### PR TITLE
Chain internal errors

### PR DESCRIPTION
**Pull request by lionel-**
Branched from #6473.

- Internal errors are chained to contextual errors.
- Group context is consistently mentioned in the contextual error, in shorter form.

Before:

```r
iris %>% group_by(Species) %>% filter(1:n())
#> Error in `filter()`:
#> ! Problem while computing `..1 = 1:n()`.
#> ✖ Input `..1` must be a logical vector, not a integer.
#> ℹ The error occurred in group 1: `Species = setosa`.
```

After:

```r
iris %>% group_by(Species) %>% filter(1:n())
#> Error in `filter()`:
#> ! Problem while computing `..1 = 1:n()`.
#> ℹ In group 1: `Species = setosa`.
#> Caused by error:
#> ! `..1` must be a logical vector, not an integer vector.
```

### CODE REVIEW

**Comment by lionel-** [comment]
I'm tempted to change:

```r
#> Error in `filter()`:
#> ! Problem while computing `..1 = 1:n()`.
#> ℹ In group 1: `Species = setosa`.
```

to the shorter:

```r
#> Error in `filter()`:
#> ! Can't compute `..1 = 1:n()`.
#> ℹ In group 1: `Species = setosa`.
```

**Review by hadley — APPROVED** [review]

**Comment by hadley** [comment]
Can't compute sounds ok to me.

File: R/mutate.R [hunk 1]
```diff
@@ -445,8 +437,9 @@ mutate_bullets <- function(cnd, ...) {
 }
 #' @export
 mutate_bullets.default <- function(cnd, ...) {
-  c(i = cnd_bullet_cur_group_label())
+  chr()
```
[the remark this answers is not in the document]
Comment (DavisVaughan) [reply R/mutate.R:440]: Same question about other default methods in summarise/filter

[the remark this answers is not in the document]
Comment (lionel-) [reply R/mutate.R:440]: We can remove them. This API used to mimick header/body/footer for conditions, and "bullets" corresponded to "body". So it made sense to have a condition with a header but no body. But now it's an interface for all the lines and it's passed to `message`, so an empty default no longer makes sense.

File: R/context.R [hunk 2]
```diff
@@ -102,7 +102,11 @@ cur_group_label <- function(type = mask_type(),
   switch(
     type,
     ungrouped = "",
-    grouped = glue("group {id}: {label}", label = group_labels_details(group)),
+    grouped = if (nrow(group)) {
+      glue("group {id}: {label}", label = group_labels_details(group))
+    } else {
+      ""
```
[the remark this answers is not in the document]
Comment (lionel-) [reply R/context.R:109]: I can no longer reproduce the error I was getting, removed.

**Pull request merged**